Environment-Definitions/TerraformTemplates/FunctionApp/environment.yaml

```diff
@@ -0,0 +1,22 @@
+name: FunctionApp
+summary: Creates a function app
+description: Deploys an environment using Terraform containing a Function App
+runner: terraform
+templatePath: function_app.tf
+parameters: 
+- id: "resource_name"
```

Comment:
We should be passing [this variable](https://github.com/Azure/deployment-environments/blob/custom-runner-private-preview/Runner-Images/Terraform/scripts/deploy.sh#L27C1-L32C57) in already. Do we want to just use that? Same with location below.